### ISSUE TITLE
Make reporter compatible with WebdriverIO v5

### ISSUE THREAD

**Issue by goatsy** [browserstack/wdio-browserstack-reporter#4]
WebdriverIO released v5 around Christmas, see also  (https://webdriver.io/blog/2018/12/19/webdriverio-v5-released.html). 

To make this reporter usable with the new version there are some small adjustments necessary. There are some minor changes that need to be made. You can find a general overview 
(https://webdriver.io/docs/customreporter.html). 

Most important ones to point out provided on this link:
https://github.com/fijijavis/wdio-mochawesome-reporter/issues/36

Thank you a lot!

**Comment by christian-bromann** [comment]
We moved this package in house, no need for support anymore. This package can be deprecated.

**Comment by ghost** [comment]
What configuration is required in order to use the reporter in v5?

When we remove that package, this error now appears:
"Couldn't find plugin "browserstack" reporter, neither as wdio scoped package "@wdio/browserstack-reporter" nor as community package "wdio-browserstack-reporter". Please make sure you have it installed!"

**Comment by christian-bromann** [comment]
instead of using `wdio-browserstack-service`, install `@wdio/browserstack-service`

**Comment by ghost** [comment]
This is the reporter package I'm concerned with. 

I'm able to successfully run tests, but not able to publish results to the tab in my build pipeline within Azure DevOps: https://www.browserstack.com/automate/azure the section on WebDriverIO.

**Comment by ckzgraphics** [comment]
TeamCity + WebDriverIO v5 - https://github.com/sullenor/wdio-teamcity-reporter/issues/12

**Comment by nbabii** [comment]
I have the same issue with BS reporter package.
I'm able to successfully run tests, but not able to publish results to the tab in my build pipeline within Azure DevOps: https://www.browserstack.com/automate/azure the section on WebDriverIO.

**Comment by ghost** [comment]
Still waiting on compatibility with v5...

**Comment by christian-bromann** [comment]
@sschweigert11 see my comment above, we moved the service to the wdio org and it is compatible with v5: https://www.npmjs.com/package/@wdio/browserstack-service

**Comment by ghost** [comment]
@christian-bromann : Thanks for the follow-up. 

What is still unclear to me is how to properly configure my configuration file to properly display results in Azure DevOps. I've configured my user, key, services: ["browserstack"] but I'm not clear on how to reference the specific reporter as it was previously reporter: ["browserstack"] in v4, but fails in v5.

**Comment by nbabii** [comment]
@christian-bromann @sschweigert11 Looks like some misunderstanding is here. This issue is related to wdio-browserstack-reporter (for wdio v4 it is available here https://www.npmjs.com/package/wdio-browserstack-reporter), as for me browserstack-service is a different component, but I may be wrong.

**Comment by OlegDubrovskyi** [comment]
@christian-bromann hello, could you please provide information how use reporter wiht wdio v5.
As I understood, browserstack-reporter moved to browserstack-service, but please share information or config, how use reporter in v5.

**Comment by christian-bromann** [comment]
@OlegDubrovskyi see https://webdriver.io/docs/browserstack-service.html

**Comment by OlegDubrovskyi** [comment]
@christian-bromann yes, but I mean browserstack reporter, in v4 I just added reporters: ['browserstack']. It's doesn't work for v5. I don't see how use browserstack reporters in webdriverio documentation and by your link

**Comment by christian-bromann** [comment]
> I don't see how use browserstack reporters in webdriverio documentation and by your link

You can not use this reporter with v5.

**Comment by ghost** [comment]
@christian-bromann : What service should we use to interface test results in v5 with BrowserStack?

**Comment by christian-bromann** [comment]
As I mentioned above: https://webdriver.io/docs/browserstack-service.html

**Comment by Alona-T** [comment]
@christian-bromann 
I am facing the same issue with v6
I tried to use browserstack-service however it does not return any report and to integrate with azure we need a reporter
If browserstack-service is also used for reporting for v6 - is it possible to see how exactly because in https://webdriver.io/docs/browserstack-service.html it is not described.

**Comment by rostrovsky** [comment]
Code from branch `wdio5` works with webdriverio v6 (I have tested it with `6.3.0`,). It just provides XML file with `sessionId`, but it's a good start if you want to download session logs via BS API.
I have raised separate issue (#10) to merge it to release it on npmjs.com

@christian-bromann - if https://webdriver.io/docs/browserstack-service.html can provide a report, could you please describe how to configure it? As of now we need to use `browserstack/wdio-browserstack-reporter` for that.

**Comment by kuda1992** [comment]
@christian-bromann  why are you repeating the same answer the questions are clear we need a reporter?

**Comment by christian-bromann** [comment]
@kuda1992 there is a BS integration with WebdriverIO called `@wdio/browserstack-service` which I recommend to use. If this service doesn't provide the features you are looking for feel free to make a feature suggestion. I am not maintaining this reporter and also don't know what it does.

**Comment by wati123321** [comment]
@rostrovsky @Alona-T BrowserStack has already started work on releasing wdio5/6 version to npmjs. In case of any further doubts, you can reach out to support@browserstack.com.
